### ISSUE TITLE
Add class "AirDribble" and GameSituation "AirDribbleSetup1"

### ISSUE THREAD

**Issue by John-WL** [John-WL/RocketLeague-PanBot#13]

**Comment by John-WL** [comment]
Added. Currently implementing

**Comment by John-WL** [comment]
Implemented, but it's badly made and unpredictable in a lot of cases.

Who knew? Turns out air dribbling is hard to code.

I need to change the part where the player's direction is handled.

I have no clue for the moment how to solve this.

Basically, I need to find an acceleration orientation that's going to move my current speed towards an arbitrary desired speed.

**Comment by John-WL** [comment]
I think I figured out what yo do.

Basically, instead of the previous code that's extremely arbitrary and unclear on why the process is what it is, I implemented a way better and reliable code that makes sense AND that has no (or almost no) arbitrary values, wich makes it way easier to tweak and adjust to my needs.

So, what I'm doing is:

1) find the point on the ball where the player needs to aim to be able to reach the trajectory;

2) find player's orientation and boost amount to reach the point on the ball.

The player direction is just the point on the ball, plus a vector that's computed from the pid of the car's position from the ball...


Wait, should I not compute the car's speed.......?

Gotta check that out...

It's not working right now, but it's way better than me already.

I need to find a way to make it even better if I want to be able to control the ball's position.

**Comment by John-WL** [comment]
Ok, I implemented the speed version, and it seem better right away.

The sad thing is that I needed to add arbitrary scalars to make it work...

And the roll actualtion is actually quite bad.

I originaly planned to actuate the roll output from an arbitrary up vector, but it doesn't seem to be working at all.

It's like the bot is bumping the ball a little bit too much if it gets under it...

Which is why I thought about updating it from the ball, but it doesn't work at all actually...

Maybe I should try to update it from the player destination?

I mean, the point on the ball has an orientation... I think the player's roll orientation is the same as the x-y coordinate of the player's destination from the ball.

Basically, I should try to take the yellow vector's x-y coordinate, and make the player's roll orientation point in that direction.

I think that would work better. Gotta try it.
